### PR TITLE
messageOnly: support bytes address for non-evm chains

### PR DESCRIPTION
**Pull request by xiaozhou**

### CODE REVIEW

File: contracts/message/libraries/MsgDataTypes.sol [hunk 1]
```diff
@@ -57,6 +57,21 @@ library MsgDataTypes {
...
     }
 
+    struct RouteInfo2 {
+        bytes sender;
+        address receiver;
+        uint64 srcChainId;
+        bytes32 srcTxHash; // src chain msg tx hash
+    }
+
+    struct RouteInfoX {
```
Comment (i9) [inline contracts/message/libraries/MsgDataTypes.sol:67]: why do we need this?

**Review by i9 — APPROVED** [review]

**Review by ClaudeZsb — APPROVED** [review]

**Review by patrickmao93 — APPROVED** [review]

**Comment by patrickmao93** [comment]
probably should merge https://github.com/celer-network/sgn-v2-contracts/pull/169 before this one

File: contracts/message/messagebus/MessageBusReceiver.sol [hunk 2]
```diff
@@ -412,20 +440,34 @@ contract MessageBusReceiver is Ownable {
...
+                    _route.srcChainId,
+                    _message,
+                    msg.sender
+                )
+            );
+        } else {
+            (ok, res) = address(_route.receiver).call{value: msg.value}(
+                abi.encodeWithSelector(
+                    bytes4(keccak256(bytes("executeMessage(bytes,uint64,bytes,address)"))),
+                    _route.sender,
```
Comment (luohaha) [inline contracts/message/messagebus/MessageBusReceiver.sol:464]: should use '_route.senderBytes' instead of '_route.sender'?

Comment (xiaozhou) [reply contracts/message/messagebus/MessageBusReceiver.sol]: fixed

File: contracts/message/libraries/MsgDataTypes.sol [hunk 3]
```diff
@@ -57,6 +57,23 @@ library MsgDataTypes {
         bytes32 srcTxHash; // src chain msg tx hash
     }
 
+    // used for msg from non-evm chains with logner-bytes address
```
Comment (Dominator008) [inline contracts/message/libraries/MsgDataTypes.sol:60]: Typo, s/logner/longer

File: contracts/message/messagebus/MessageBusReceiver.sol [hunk 4]
```diff
@@ -459,6 +501,14 @@ contract MessageBusReceiver is Ownable {
         return abi.decode(_returnData, (string)); // All that remains is the revert string
     }
 
+    function getRoutInfo(MsgDataTypes.RouteInfo calldata _route) private pure returns (MsgDataTypes.Route memory) {
```
Comment (Dominator008) [inline contracts/message/messagebus/MessageBusReceiver.sol:504]: s/getRoutInfo/getRouteInfo

Comment (xiaozhou) [reply contracts/message/messagebus/MessageBusReceiver.sol]: fixed

File: contracts/message/libraries/MsgDataTypes.sol (hunk 3 above)
Comment (xiaozhou) [reply contracts/message/libraries/MsgDataTypes.sol]: fixed

**Review by Dominator008 — APPROVED** [review]

**Pull request merged**